activejob/lib/active_job/instrumentation.rb

```diff
@@ -18,12 +18,17 @@ module Instrumentation #:nodoc:
     private
       def instrument(operation, payload = {}, &block)
         enhanced_block = ->(event_payload) do
-          aborted = !block.call if block
-          event_payload[:aborted] = true if aborted
+          block.call if block
+          event_payload[:aborted] = true if defined?(@_halted_callback_hook_called) && @_halted_callback_hook_called
         end
 
         ActiveSupport::Notifications.instrument \
           "#{operation}.active_job", payload.merge(adapter: queue_adapter, job: self), &enhanced_block
       end
+
+      def halted_callback_hook(_)
+        super
+        @_halted_callback_hook_called = true
```

Comment:
I think we need to find a place to reset this value otherwise when we try to run or enqueue the job in the same instance `_halted_callback_hook_called` will be `true`.

Reply:
👍 done